llvm-general/src/LLVM/General/Internal/Module.hs

```diff
@@ -236,6 +236,19 @@ getDataLayout m = do
   dlString <- decodeM =<< FFI.getDataLayout m
   either fail return . runExcept . parseDataLayout A.BigEndian $ dlString
 
+-- | Build an LLVM.General.'Module' from a
+-- LLVM.General.AST.'LLVM.General.AST.Module' with a 'DataLayout'
+-- appropriate for the given 'TargetMachine'. This is important for
+-- native code generation as it impacts name mangling.
+withModuleFromASTForTargetMachine :: Context -> TargetMachine -> A.Module
```

Comment:
Does this function have any equivalent in LLVM? The space of potential helper functions like this is too big. If we added everything of equivalent value we'd never finish, and if we don't we get an inconsistent patchwork. I've chosen to resolve that choice by sticking as closely as I can to stuff-that-exists-in-LLVM.